sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -751,3 +751,17 @@ def test_dist_threshold_invalid_parameters():
         AgglomerativeClustering(n_clusters=None,
                                 distance_threshold=1,
                                 compute_full_tree=False).fit(X)
+
+
+def test_invalid_shape_precomputed_dist_matrix():
+    rng = np.random.RandomState(0)
+    # Test non square distance matrix
```

Comment:
Can you move the comment at the top of the function with a little bit more details like
```
# Check that an error is raise when affinity='precomputed' and a non square matrix is passed (PR #16257).
```

Reply:
Ok, done.